### PR TITLE
rohd-hcl: parity

### PR DESCRIPTION
**Pull request by RPG-coder-intc**
## Description & Motivation

Note: Requires Find PR #37  to be merged in before this PR.

A modules for adding functionality of Parity.

Contains 2 modules for parity functionality:
1. ParityTransmitter Module - Appends parity bits to given data that can be used for transmission
2. ParityReceiver Module - 
    - Takes the transmitted data with parity bit. 
    - Splits the data into original data and parity bit. 
    - Also checks for parity error (result) within original data

## Related Issue(s)

Related to issue #39 

## Testing

Testing is handled within `/test/parity_test.dart`

## Backwards-compatibility

Yes, this PR is backward compatible. This is an added functionality. 
Prior modules are not unaffected or reused for completion of this functionality.

## Documentation

Documentation is handled.

### CODE REVIEW

**Comment by RPG-coder-intc** [comment]
@mkorbel1 this PR is ready for a review. also note: this PR contains files from #37

**Review by mkorbel1 — CHANGES_REQUESTED** [review]
Looks good!  Just some minor tweaks

File: lib/src/parity.dart [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+  /// [_output] is output of parity (use index for accessing from outside Module)
+  late Logic _output;
+
+  /// [data] is an getter for transmit data having a parity check
+  Logic get data => _output;
+
+  /// Construct a module
+  ParityTransmitter(Logic bus) {
+    bus = addInput('bus', bus, width: bus.width);
+    final transmitData = [bus, bus.xor()].swizzle();
```
[the remark this answers is not in the document]
Comment (RPG-coder-intc) [reply lib/src/parity.dart]: done

File: doc/parity.md [hunk 2]
```diff
@@ -0,0 +1,11 @@
+# Parity
+
+ROHD HCL implements `Parity` functionality class that other modules can extend for error checking. For satisfying the functionality of Parity error checking in `Logic` data, ROHD HCL provides 2 `Module`, namely `ParityTransmitter` and `ParityReceiver`.
```
[the remark this answers is not in the document]
Comment (RPG-coder-intc) [reply doc/parity.md]: done

File: lib/src/parity.dart [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+
+/// Encode data to transport with Parity bits
+class ParityTransmitter extends Module {
+  /// [_output] is output of parity (use index for accessing from outside Module)
+  late Logic _output;
+
+  /// [data] is an getter for transmit data having a parity check
+  Logic get data => _output;
+
+  /// Construct a module
```
[the remark this answers is not in the document]
Comment (RPG-coder-intc) [reply lib/src/parity.dart]: done

**Review by mkorbel1 — APPROVED** [review]
Looks good to me, nice work!  Once he find/count stuff merges in, we can merge this.

Be sure to merge in that branch's new fixes into this branch to avoid any merge conflicts.

**Review by mkorbel1 — APPROVED** [review]

**Review by mkorbel1 — APPROVED** [review]

**Comment by RPG-coder-intc** [comment]
@mkorbel1 @chaparalas @quekyj  can this pr be merged in :)

**Comment by mkorbel1** [comment]
Sorry for the delay!

**Comment by RPG-coder-intc** [comment]
> Sorry for the delay!

No worries :) Thanks...

**Pull request merged**